src/Plugin/StrawberryRunnersPostProcessor/OcrPostProcessor.php

```diff
@@ -377,8 +440,129 @@ protected function hOCRtoMiniOCR($output, $pageid) {
+    // First pdf2djvu -q --no-metadata -p 2 -j0 -o some_output_file.djv %file
+    // Second djvudump some_output_file.djv |grep TXTz |wc -l
+
+    $command = '';
+    $can_run_pdf2djvu = \Drupal::service('strawberryfield.utility')
+      ->verifyCommand($execpath_pdf2djvu);
+    $can_run_djvudump = \Drupal::service('strawberryfield.utility')
+      ->verifyCommand($execpath_djvudump);
+    $filename = pathinfo($file_path, PATHINFO_FILENAME);
+    $sourcefolder = pathinfo($file_path, PATHINFO_DIRNAME);
+    $sourcefolder = strlen($sourcefolder) > 0 ? $sourcefolder . '/' : sys_get_temp_dir() . '/';
+    $pdf2djvu_destination_filename = "{$sourcefolder}{$filename}_{$page_number}.djv";
+    if ($can_run_pdf2djvu &&
+      $can_run_djvudump &&
+      (strpos($arguments_pdf2djvu, '%file') !== FALSE) &&
+      (strpos($arguments_djvudump, '%file') !== FALSE)) {
+      $arguments_pdf2djvu = "-q --no-metadata -j0 -p {$page_number} -o $pdf2djvu_destination_filename " . $arguments_pdf2djvu;
+      $arguments_pdf2djvu = str_replace('%s', '', $arguments_pdf2djvu);
+      $arguments_pdf2djvu = str_replace_first('%file', '%s', $arguments_pdf2djvu);
+      $arguments_pdf2djvu = sprintf($arguments_pdf2djvu, $file_path);
+
+      $arguments_djvudump = str_replace('%s', '', $arguments_djvudump);
+      $arguments_djvudump = str_replace_first('%file', '%s', $arguments_djvudump);
+      $arguments_djvudump = sprintf($arguments_djvudump, $pdf2djvu_destination_filename);
+
+      $command_pdf2djvu = escapeshellcmd($execpath_pdf2djvu . ' ' . $arguments_pdf2djvu);
+      $command_djvudump = escapeshellcmd($execpath_djvudump . ' ' . $arguments_djvudump);
+
+      $command = $command_pdf2djvu . ' && ' . $command_djvudump . ' |grep TXTz |wc -l';
 
+    }
+    else {
+      error_log("missing arguments for OCR");
+    }
+    // Only return $command if it contains the original filepath somewhere
+    if (strpos($command, $file_path) !== FALSE) {
+      return $command;
+    }
+    return '';
 
+  }
 
+  /**
+   * Builds a clean Command string using a File path.
+   *
+   * @param \stdClass $io
+   *    $io->input needs to contain
+   *           \Drupal\strawberry_runners\Annotation\StrawberryRunnersPostProcessor::$input_property
+   *           \Drupal\strawberry_runners\Annotation\StrawberryRunnersPostProcessor::$input_arguments
+   *    $io->output will contain the result of the processor
+   *
+   * @return null|string
+   */
+  public function buildExecutableCommand_djvu2hocr(\stdClass $io) {
+    $input_property = $this->pluginDefinition['input_property'];
+    $input_argument = $this->pluginDefinition['input_argument'];
+    // Sets the default page to 1 if not passed.
+    $file_path = isset($io->input->{$input_property}) ? $io->input->{$input_property} : NULL;
+    $page_number = isset($io->input->{$input_argument}) ? (int) $io->input->{$input_argument} : 1;
+    $config = $this->getConfiguration();
+    $execpath_djvu2hocr = '/usr/bin/djvu2hocr';
```

Comment:
Should I add this to the config?

Reply:
Not only this one, also the config for pdf2djvu and djvudump you can find in function buildExecutableCommand_checkSearchable above.

Reply:
Obviously if you agree with code, so before add to config check code and make everything you feel to changed/refactor.